tests/flavor_test.go

```diff
@@ -264,7 +315,170 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			Expect(vmi.Annotations[v1.ClusterPreferenceAnnotation]).To(Equal(""))
 		})
 
-		It("[test_id:TODO] should fail if flavor and VM define CPU", func() {
+		It("[test_id:TODO] should apply VirtualMachineFlavor and VirtualMachinePreference from VirtualMachineInstance", func() {
+			vmi := tests.NewRandomVMIWithEphemeralDisk(
+				cd.ContainerDiskFor(cd.ContainerDiskCirros),
+			)
+
+			flavor := newVirtualMachineFlavor(vmi)
+			flavor, err = virtClient.VirtualMachineFlavor(util.NamespaceTestDefault).
+				Create(context.Background(), flavor, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			preference := newVirtualMachinePreference(&flavorv1alpha1.VirtualMachinePreferenceSpec{
+				CPU: &flavorv1alpha1.CPUPreferences{
+					PreferredCPUTopology: flavorv1alpha1.PreferSockets,
+				},
+				Devices: &flavorv1alpha1.DevicePreferences{
+					PreferredDiskBus: v1.DiskBusSATA,
+				},
+				Features: &flavorv1alpha1.FeaturePreferences{
+					PreferredHyperv: &v1.FeatureHyperv{
+						VAPIC: &v1.FeatureState{
+							Enabled: pointer.Bool(true),
+						},
+						Relaxed: &v1.FeatureState{
+							Enabled: pointer.Bool(true),
+						},
+					},
+				},
+				Firmware: &flavorv1alpha1.FirmwarePreferences{
+					PreferredUseBios: pointer.Bool(true),
+				},
+				// We don't want to break tests randomly so just use the q35 alias for now
+				Machine: &flavorv1alpha1.MachinePreferences{
+					PreferredMachineType: "q35",
+				},
+			})
+
+			preference, err = virtClient.VirtualMachinePreference(util.NamespaceTestDefault).
+				Create(context.Background(), preference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			// Remove any requested resources from the VMI
+			removeResourcesAndPreferencesFromVMI(vmi)
+
+			// Add the flavor and preference matchers to the VMI spec
+			vmi.Spec.Flavor = &v1.FlavorMatcher{
+				Name: flavor.Name,
+				Kind: flavorapi.SingularResourceName,
+			}
+			vmi.Spec.Preference = &v1.PreferenceMatcher{
+				Name: preference.Name,
+				Kind: flavorapi.SingularPreferenceResourceName,
+			}
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Observe the VirtualMachineInstance created
+			Eventually(func() error {
+				_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				return err
+			}, 300*time.Second, 1*time.Second).Should(Succeed())
```

Comment:
Is this block needed? When `Create()` call succeeds, the VMI should already exist.

Reply:
ACK